### ISSUE TITLE
False positive for UnnecessaryParentheses

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#836]
The following code gets flagged:

```kotlin
private fun isTypeThatRequiresAnnotation(psiType: PsiType): Boolean {
  val canonicalText = psiType.canonicalText
      .replace("<[\\w.]*>".toRegex(), "") // We need to remove the generics.

  return (canonicalText.matches("io\\.reactivex\\.[\\w]+".toRegex()) ||
      "io.reactivex.disposables.Disposable" == canonicalText ||
      "io.reactivex.observers.TestObserver" == canonicalText ||
      "io.reactivex.subscribers.TestSubscriber" == canonicalText)
}
```

The `(..)` around the first expression after the return statement are mandatory and hence Detekt should not flag this.

**Comment by arturbosch** [comment]
Even intellij suggests to remove the unnecessary parentheses:

```kotlin
	return canonicalText.matches("io\\.reactivex\\.[\\w]+".toRegex()) ||
			"io.reactivex.disposables.Disposable" == canonicalText ||
			"io.reactivex.observers.TestObserver" == canonicalText ||
			"io.reactivex.subscribers.TestSubscriber" == canonicalText
```

**Comment by vanniktech** [comment]
Whops. Yes. I totally missed those ones.
